src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -53,9 +53,11 @@ const { bech32 } = require('cardano-crypto.js')
 
 const TransportNodeHid = require('@ledgerhq/hw-transport-node-hid').default
 
-export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
-  const transport = await TransportNodeHid.create()
-  const ledger = new Ledger(transport)
+export const LedgerCryptoProvider: (transport?: any) => Promise<CryptoProvider> = async (transport?) => {
+  const transportToUse = (transport != null)
+    ? transport
+    : await TransportNodeHid.create()
+  const ledger = new Ledger(transportToUse)
```

Comment:
Could we maybe get rid of the `null`/`undefined` case here? All `LedgerCryptoProvider()` calls use `getTransport()` (which always returns one of the transports), except for the call in `commandExecutor.ts`, where we could call `LedgerCryptoProvider(await TransportNodeHid.create())` directly.

Also, isn't there a Transport interface provided by Ledger (that we could use instead of `any`)?

Reply:
Yes, there is, I just failed to import it properly at the time... and forgot to go back to it. It should be fixed now.